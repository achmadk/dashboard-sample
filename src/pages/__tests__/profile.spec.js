import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import chai from 'chai'

import ProfilePageAsyncContainer, { profileComponent } from '@/pages/profile'
import ProfilePageContainer from '@/pages/profile/profile'

const props = {
  enqueueSnackbar: jest.fn()
}

describe('Profile Page Async Container', () => {
  it('render correctly', () => {
    const wrapper = shallow(<ProfilePageAsyncContainer {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('return profile component successfully', () => {
    const result = profileComponent()
    chai.expect(result).to.exist
  })
})

describe('Profile Page Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(<ProfilePageContainer {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
