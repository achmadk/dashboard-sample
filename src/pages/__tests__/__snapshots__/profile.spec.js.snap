// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page Async Container render correctly 1`] = `
<React.Suspense
  fallback={
    <div>
      Loading...
    </div>
  }
>
  <React.Lazy />
</React.Suspense>
`;

exports[`Profile Page Component render correctly 1`] = `
<span>
  Profile
</span>
`;
